docs/metrics-data-model.md

```diff
@@ -0,0 +1,177 @@
+# k6 Time Series
+
+We would migrate to a time series data model for getting the benefits of aggregated data so just allocating samples one time and pass around the aggregated time series value (or values - e.g. in case of Trend). The Outputs will switch from fetching the buffered samples to fetching a snapshot of the time series (the same concept adopted by Prometheus' scraping operation).
+
+## Data model
+
+#### Metrics and Series
+
+* Metric (name, type)
+* TimeSeries (metric, tagset)
+* Sample (metric, tagset, timestamp, value)
+* TimeSeries Sample (timeseries ref, timestamp, value)
+
+```go
+// This is the current k6 Metric polished from the Thresholds stuff.
+type Metric struct {
+	Name     string     `json:"name"`
+	Type     MetricType `json:"type"`
+	Contains ValueType  `json:"contains"`
+}
+
+type TimeSeries struct {
+    ID uin64 // hash(metric+tags)
+	MetricName string
+	Tags   TagSet
+
+	Sink Sink // Counter, Guage, Rate, Trend (SparseHistogram in the future)
+}
+
+type Sink interface {
+    AddPoint(v float64)
+}
+
+type TSample struct { 
+	TimeSeriesID string
+	Timestamp      uint64
+	Value          float64
+}
+
+type Sample struct {
+        MetricName string
+        Timestamp uint64
+        Tags TagSet
+        Value float64
+}
+```
+
+#### Open questions:
+
+* `TimeseriesSample` knows the time series concept and it has the reference to a time series. The Sample is raw data and there isn't yet an identified/matched time series to reference. In this way, the collecting phase doesn't  require knowledge from a time series database. It could speed up the  implementation and introduction of the first stage, where hopefully we don't touch the sample generation (e.g. all the `k6/js/modules`).
+* In the case the two types (`Sample` and `TSample`) are confirmed, we need the `TagSet` in the Sample or can we just simplify it as a `[]string` or `map[string]string` and delegate it in a centralized place the `TagSet` association?
+* Currently, the metric type is defined by the `metrics.Metric` but the same type handle Thresholds and Sink operation. Should the `TimeSeries` type depend on `Metric`? Or should we wait to split them in a different struct?
+
```

Comment:
```suggestion
* Do we need `Contains ValueType`?
```
It seems used only from the summary. Should we move the mapping directly there? If not, then I think we should extend the [current JS API](https://k6.io/docs/javascript-api/k6-metrics/trend/) for supporting all the possible values (currently, it supports only Time).